client/src/actions/grantsMetadata.ts

```diff
@@ -0,0 +1,126 @@
+import { ethers } from "ethers";
+import { Dispatch } from "redux";
+import { RootState } from "../reducers";
+import { Metadata } from "../types";
+import { global } from "../global";
+import { addressesByChainID } from "../contracts/deployments";
+import GrantsRegistryABI from "../contracts/abis/GrantsRegistry.json";
+
+export const GRANT_METADATA_LOADING_URI = "GRANT_METADATA_LOADING_URI";
+export interface GrantMetadataLoadingURI {
+  type: typeof GRANT_METADATA_LOADING_URI;
+  id: number;
+}
+
+export const GRANT_METADATA_LOADING = "GRANT_METADATA_LOADING";
+export interface GrantMetadataLoading {
+  type: typeof GRANT_METADATA_LOADING;
+  id: number;
+}
+
+export const GRANT_METADATA_FETCHED = "GRANT_METADATA_FETCHED";
+export interface GrantMetadataFetched {
+  type: typeof GRANT_METADATA_FETCHED;
+  data: Metadata;
+}
+
+export type GrantMetadataActions =
+  | GrantMetadataLoadingURI
+  | GrantMetadataLoading
+  | GrantMetadataFetched;
+
+export const grantMetadataLoadingURI = (id: number): GrantMetadataActions => ({
+  type: GRANT_METADATA_LOADING_URI,
+  id,
+});
+
+export const grantMetadataLoading = (id: number): GrantMetadataActions => ({
+  type: GRANT_METADATA_LOADING,
+  id,
+});
+
+export const grantMetadataFetched = (data: Metadata): GrantMetadataActions => ({
+  type: GRANT_METADATA_FETCHED,
+  data,
+});
+
+export const fetchGrantData =
+  (id: number) => async (dispatch: Dispatch, getState: () => RootState) => {
+    dispatch(grantMetadataLoadingURI(id));
+    const state = getState();
+    const { chainID } = state.web3;
+    const addresses = addressesByChainID(chainID!);
+    const signer = global.web3Provider?.getSigner();
+
+    const grantRegistry = new ethers.Contract(
+      addresses.grantsRegistry,
+      GrantsRegistryABI,
+      signer
+    );
+
+    let grant: { metadata: string };
+    try {
+      grant = await grantRegistry.grants(id);
```

Comment:
Maybe we can consolidate these in the future? Or make functions for related requests